plugins/c8y_firmware_plugin/src/lib.rs

```diff
@@ -94,15 +94,7 @@ async fn run_with(
 
     // Create actor instances
     let mqtt_config = tedge_config.mqtt_config()?;
-    let mut jwt_actor = C8YJwtRetriever::builder(
-        mqtt_config.clone(),
-        tedge_config
-            .c8y
-            .try_get(c8y_profile)?
-            .bridge
-            .topic_prefix
-            .clone(),
-    );
+    let mut jwt_actor = C8YHeaderRetriever::try_builder(&tedge_config, c8y_profile)?;
```

Comment:
```suggestion
    let mut auth_actor = C8YHeaderRetriever::try_builder(&tedge_config, c8y_profile)?;
```

Reply:
I'll not change code in `c8y_firmware_plugin` unless it's critical since it's a legacy feature.